src/server/store.rs

```diff
@@ -1,51 +1,320 @@
+
+        // Insert new block reference into index.
+        self.index.insert(block.start_timestamp.unwrap().timestamp_millis(), block_filename);
+
+        // Rewrite index to disk.
+        let index_filename = format!("{}/index.rdb", dotenv::var("DATAROOT").unwrap());
+        fs::write(&index_filename, bincode::serialize(&self.index).unwrap()).expect("ERROR: writing index to disk");
+
+        // Flush block from memory.
+        block.flush();
+        println!("Flushed block to disk.");
+    }
+
+    // Get all blocked in packed form.
+    pub fn get_packed_blocks(&self) -> Vec<PackedBlock> {
+        // For each block in the index...
+        let mut ret = vec![];
+        for (_, v) in self.index.iter() {
+            // Unpack the given file.
+            let packed_block = PackedBlock::from_filepath(v.clone());
+            ret.push(packed_block);
+        }
+        ret
+    }
+
+    // Get all blocks in a certain time range (by start times) in packed form.
+    pub fn get_packed_blocked_range(&self, start_timestamp: DateTime<Utc>, end_timestamp: DateTime<Utc>) -> Vec<PackedBlock> {
+        // For each block in the index...
+        let mut ret = vec![];
+        for (_, v) in self.index.range(start_timestamp.timestamp_millis()..end_timestamp.timestamp_millis()) {
+            // Unpack the given file.
+            let packed_block = PackedBlock::from_filepath(v.clone());
+            ret.push(packed_block);
+        }
+        ret
+    }
+}
 
 // Block Struct.
 pub struct Block {
-    // TODO: Get this to work with FSTs
-    // TODO: Do we want this index to work over both labels and metrics?
-    //      We should probably split it up...?
-    // pub index: fst::MapBuilder<RoaringBitmap>,
-    index: HashMap<String, RoaringBitmap>,
+    index: HashMap<String, Bitmap>,
     storage: Vec<Series>,
     id_map: Vec<String>,
     key_map: HashMap<String, usize>,
+    start_timestamp: Option<DateTime<Utc>>,
+    end_timestamp: Option<DateTime<Utc>>,
+    frozen: bool,
 }
-
 impl Block {
+    // Constructor.
     pub fn new() -> Self {
         Block {
-            // index: fst::MapBuilder::memory(),
             index: HashMap::new(),
             storage: vec![],
             id_map: vec![],
             key_map: HashMap::new(),
+            start_timestamp: Some(Utc::now()),
+            end_timestamp: None,
+            frozen: false,
         }
     }
 
+    // Return series.
     pub fn get_storage(&self) -> &Vec<Series> {
         &self.storage
     }
 
-    pub fn search_index(&self, key: String) -> Option<&RoaringBitmap> {
+    // Get the bitmap for a specific label / metric.
+    pub fn search_index(&self, key: String) -> Option<&Bitmap> {
         self.index.get(&key)
     }
+
+    // Convert a block to bytes.
+    pub fn to_bytes(&mut self) -> Vec<u8> {
+        // Freeze and cap the block.
+        self.frozen = true;
+        self.end_timestamp = Some(Utc::now());
```

Comment:
I think we can get away with it for now, but we should really use the timestamps of the datapoints that we'll be searching through rather than the timestamp of our current system. Made a note of this in #17